src/server/game/Entities/Player/Player.cpp

```diff
@@ -15098,17 +15098,34 @@ void Player::FailQuest(uint32 questId)
             SendQuestFailed(questId);
 
         // Destroy quest items on quest failure.
-        for (uint8 i = 0; i < QUEST_OBJECTIVES_COUNT; ++i)
+        for (uint8 i = 0; i < QUEST_SOURCE_ITEM_IDS_COUNT; ++i)
```

Comment:
That was my bad. I had an old branch, and forgot that I had already found that I was wrong. However, QUEST_ITEM_OBJECTIVES_COUNT is correct: https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/game/Quests/QuestDef.h#L283